Drone/drone_base.py

```diff
@@ -0,0 +1,361 @@
+        the drone's barometer
+
+        Returns:
+        ----------
+        double
+            The distance from the ground
+        """
+        return self.vehicle.location.global_relative_frame.alt
+
+    def arm(self, timeout = 60):
+        """
+        Attempts to arm the drone for flight
+
+        Parameters
+        ----------
+        timeout: Integer (optional)
+            The duration in seconds that arming should be attempted before
+            giving up and returning
+
+        Precondition:
+        ----------
+        self.connect has been called and succeeded.
+
+        Postcondition:
+        ----------
+        Upon successful arm, self.vehicle.armed is True, and false otherwise.
+        The drone is now suitable to take off.
+
+        Returns:
+        ----------
+        None
+        """
+        start = time.time()
+        self.vehicle.mode = VehicleMode(c.GUIDED_MODE)
+
+        self.logger.info(threading.current_thread().name + ": Arming")
+        start = time.time()
+        while (not self.vehicle.armed) and (time.time() - start < timeout):
+            self.vehicle.armed = True
+            time.sleep(1)
+        if not self.vehicle.armed:
+            self.logger.error(threading.current_thread().name + ": Failed to arm")
+        else:
+            self.logger.info(threading.current_thread().name + ": Armed")
+
+    def is_connected(self):
+        """
+        Getter for self.connected
+
+        Parameters
+        ----------
+        None
+
+        Returns:
+        ----------
+        Boolean
+            True if the drone is connected to Ardupilot and false otherwise
+        """
+        return self.connected
+
+    def is_armed(self):
+        """
+        Getter for self.vehicle.armed
+
+        Parameters
+        ----------
+        None
+
+        Returns:
+        ----------
+        Boolean
+            True if the drone is armed and false otherwise
+        """
+        return self.vehicle.armed
+
+    def takeoff(self, target_altitude, stop_event):
+        """
+        The drone flies from the ground to desired altitude
+
+        Parameters
+        ----------
+        target_altitude: Integer 
+            The height in meters the drone should be off the ground after
+            this function completes
+        stop_event: threading.Event
+            Event which is set when takeoff should be canceled
+
+        Precondition:
+        ----------
+        The drone is connected and armed.
+
+        Postcondition:
+        ----------
+        Upon success: the drone is the requested altitude in the air. 
+
+        Returns:
+        ----------
+        None
+        """
+        def takeoff_thread(target_altitude, finished, stop_event):
+            self.logger.info(threading.current_thread().name + ": Starting takeoff")
+            thrust = c.DEFAULT_TAKEOFF_THRUST
+
+            start_time = time.time()
+            cutoff_time = 10
```

Comment:
```suggestion
            cutoff_time = timeout
```

Reply:
As per your previous suggestion, I gave takeoff_thread() an additional parameter for cutoff time with a default value. So I just got rid of the declaration of cutoff_time altogether since it is now an argument.